### PR TITLE
fix commit and rollback codec bug #14

### PR DESCRIPTION
**Pull request by luky116**
when received a commit or rollback message, it will appear a error log

### Which issue(s) this PR fixes:

Fixes https://github.com/seata/seata-go/issues/14

### CODE REVIEW

File: pkg/protocol/codec/branch_commit_response_codec.go [hunk 1]
```diff
@@ -31,23 +32,39 @@ type BranchCommitResponseCodec struct {
...
-	req, _ := in.(message.BranchCommitResponse)
+	data, _ := in.(message.BranchCommitResponse)
 	buf := bytes.NewByteBuffer([]byte{})
 
-	bytes.WriteString16Length(req.Xid, buf)
-	buf.WriteInt64(req.BranchId)
-	buf.WriteByte(byte(req.BranchStatus))
+	buf.WriteByte(byte(data.ResultCode))
+	if data.ResultCode == message.ResultCodeFailed {
+		var msg string
```
Comment (baerwang) [inline pkg/protocol/codec/branch_commit_response_codec.go:56]:
```go
msg := data.Msg
if len(data.Msg) > 128 {
	msg = data.Msg[:128]
}
bytes.WriteString16Length(msg, buf)
```

File: pkg/protocol/codec/branch_commit_response_codec_test.go [hunk 2]
```diff
@@ -20,6 +20,8 @@ package codec
 import (
 	"testing"
 
+	transaction2 "github.com/seata/seata-go/pkg/common/error"
```
Comment (baerwang) [inline pkg/protocol/codec/branch_commit_response_codec_test.go:23]: serror

File: pkg/protocol/codec/branch_commit_response_codec.go [hunk 3]
```diff
@@ -19,6 +19,7 @@ package codec
 
 import (
 	"github.com/seata/seata-go/pkg/common/bytes"
+	error2 "github.com/seata/seata-go/pkg/common/error"
```
Comment (baerwang) [inline pkg/protocol/codec/branch_commit_response_codec.go:22]: serror

File: pkg/protocol/codec/branch_rollback_response_codec.go [hunk 4]
```diff
@@ -19,6 +19,7 @@ package codec
 
 import (
 	"github.com/seata/seata-go/pkg/common/bytes"
+	error2 "github.com/seata/seata-go/pkg/common/error"
```
Comment (baerwang) [inline pkg/protocol/codec/branch_rollback_response_codec.go:22]: serror

File: pkg/protocol/codec/branch_rollback_response_codec.go [hunk 5]
```diff
@@ -45,6 +51,17 @@ func (g *BranchRollbackResponseCodec) Encode(in interface{}) []byte {
 	data, _ := in.(message.BranchRollbackResponse)
 	buf := bytes.NewByteBuffer([]byte{})
 
+	buf.WriteByte(byte(data.ResultCode))
+	if data.ResultCode == message.ResultCodeFailed {
```
Comment (baerwang) [inline pkg/protocol/codec/branch_rollback_response_codec.go:55]: 这块是不是可以抽出来复用？

File: pkg/remoting/processor/client/rm_branch_commit_processor.go [hunk 6]
```diff
@@ -46,22 +46,42 @@ func (f *rmBranchCommitProcessor) Process(ctx context.Context, rpcMessage messag
...
+			},
 			Xid:          xid,
 			BranchId:     branchID,
 			BranchStatus: status,
 		},
 	}
 	err = getty2.GetGettyRemotingClient().SendAsyncResponse(response)
 	if err != nil {
-		log.Error("BranchCommitResponse error: {%#v}", err.Error())
+		log.Error("send branch commit response error: {%#v}", err.Error())
```
Comment (baerwang) [inline pkg/remoting/processor/client/rm_branch_commit_processor.go:82]: errorf

**Review by 106umao — APPROVED** [review]

File: pkg/protocol/codec/branch_rollback_response_codec.go (hunk 5 above)
Comment (luky116) [reply pkg/protocol/codec/branch_rollback_response_codec.go:57]: 这块和业务关系有点大，感觉不是很好抽出来

**Review by AlexStocks — APPROVED** [review]

**Review by baerwang — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**